src/utils/request-queue.js

```diff
@@ -91,14 +91,16 @@ class RequestQueue {
 
             // Request was successful, but returned a non-2XX status code
             else {
+              const instance = this;
```

Comment:
Maybe shift this up a little in scope so that we're not using a combination of `instance.eventEmitter` and `this.eventEmitter`?